diff --git a/config.def.h b/config.def.h
index 9efa774..a8be62c 100644
--- a/config.def.h
+++ b/config.def.h
@@ -5,8 +5,8 @@ static const unsigned int borderpx  = 1;        /* border pixel of windows */
 static const unsigned int snap      = 32;       /* snap pixel */
 static const int showbar            = 1;        /* 0 means no bar */
 static const int topbar             = 1;        /* 0 means bottom bar */
-static const char *fonts[]          = { "monospace:size=10" };
-static const char dmenufont[]       = "monospace:size=10";
+static const char *fonts[]          = { "FontAwesome:size=10", "hack:size=10" };
+static const char dmenufont[]       = "hack:size=10";
 static const char col_gray1[]       = "#222222";
 static const char col_gray2[]       = "#444444";
 static const char col_gray3[]       = "#bbbbbb";
@@ -19,7 +19,7 @@ static const char *colors[][3]      = {
 };
 
 /* tagging */
-static const char *tags[] = { "1", "2", "3", "4", "5", "6", "7", "8", "9" };
+static const char *tags[] = {"", "", "", "", "" };
 
 static const Rule rules[] = {
 	/* xprop(1):
@@ -27,8 +27,21 @@ static const Rule rules[] = {
 	 *	WM_NAME(STRING) = title
 	 */
 	/* class      instance    title       tags mask     isfloating   monitor */
-	{ "Gimp",     NULL,       NULL,       0,            1,           -1 },
-	{ "Firefox",  NULL,       NULL,       1 << 8,       0,           -1 },
+	{ "Gimp",   		  NULL,       NULL,       1 << 4,       0,    		-1 },
+	{ "Abiword",  		  NULL,       NULL,       1 << 4,       0,		-1 },
+	{ "Gnumeric",  		  NULL,       NULL,       1 << 4,	0,		-1 },
+	{ "Homebank", 		  NULL,	      NULL,       1 << 4,	0,		-1 },
+	{ "scribus",		  NULL,	      NULL,	  1 << 4,	0,		-1 },
+	{ "steam",    		  NULL,	      NULL,       1 << 3,	0,		-1 },
+	{ "Lutris",    		  NULL,	      NULL,       1 << 3,	0,		-1 },
+	{ "scummvm",    	  NULL,	      NULL,       1 << 3,	0,		-1 },
+	{ "minigalaxy",	    	  NULL,       NULL,	  1 << 3,	0,		-1 },
+	{ "mpv",    		  NULL,	      NULL,       1 << 2,	0,		-1 },
+	{ "vlc",    		  NULL,       NULL,	  1 << 2,	0,		-1 },
+	{ "Firefox",	 	  NULL,       NULL,            2,       0, 	        -1 },
+	{ "geary",	 	  NULL,       NULL,            2,       0, 	        -1 },
+	{ "chrome", 		  NULL,	      NULL,            2,	0,		-1 },
+	{ "Surf",   		  NULL,	      NULL,    	       2,	0,		-1 },
 };
 
 /* layout(s) */
diff --git a/dwm.c b/dwm.c
index 1443802..2953e32 100644
--- a/dwm.c
+++ b/dwm.c
@@ -734,13 +734,12 @@ drawbar(Monitor *m)
 	x = drw_text(drw, x, 0, w, bh, lrpad / 2, m->ltsymbol, 0);
 
 	if ((w = m->ww - tw - x) > bh) {
+		drw_setscheme(drw, scheme[m == selmon ? SchemeSel : SchemeNorm]);
 		if (m->sel) {
-			drw_setscheme(drw, scheme[m == selmon ? SchemeSel : SchemeNorm]);
 			drw_text(drw, x, 0, w, bh, lrpad / 2, m->sel->name, 0);
 			if (m->sel->isfloating)
 				drw_rect(drw, x + boxs, boxs, boxw, boxw, m->sel->isfixed, 0);
 		} else {
-			drw_setscheme(drw, scheme[SchemeNorm]);
 			drw_rect(drw, x, 0, w, bh, 1, 1);
 		}
 	}
